src/rules/exports-subpaths-style.ts

```diff
@@ -0,0 +1,149 @@
+import type { AST as JsonAST } from "jsonc-eslint-parser";
+
+import { createRule } from "../createRule.ts";
+
+type ExportsValue = JsonAST.JSONObjectExpression | JsonAST.JSONStringLiteral;
+
+function hasMultipleSubpaths(node: JsonAST.JSONObjectExpression): boolean {
+	const subpaths = node.properties.filter(
+		(property) =>
+			property.key.type === "JSONLiteral" &&
+			typeof property.key.value === "string" &&
+			property.key.value.startsWith("."),
+	);
+
+	return subpaths.length > 1;
+}
+
+function isImplicitFormat(node: ExportsValue): boolean {
+	if (node.type === "JSONLiteral") {
+		return true;
+	}
+
+	const keys = node.properties
+		.filter((property) => property.key.type === "JSONLiteral" && typeof property.key.value === "string")
+		.map((property) => (property.key as JsonAST.JSONStringLiteral).value);
+
+	// If all keys are conditions (no subpaths starting with "."), it's implicit
+	// Subpaths start with "." while conditions don't (import, require, node, default, types, browser)
+	return keys.every((key) => !key.startsWith("."));
+}
+
+export const rule = createRule({
+	create(context) {
+		const [{ prefer = "explicit" } = {}] = context.options;
+
+		function validateForExplicit(node: JsonAST.JSONProperty) {
+			if (node.value.type !== "JSONLiteral" && node.value.type !== "JSONObjectExpression") {
+				return;
```

Comment:
It looks like there aren't any tests that cover this early return.  Can we add a test that covers this, or Is the condition not necessary?